React/Views/RCTModalHostView.m

```diff
@@ -89,6 +101,13 @@ - (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
   RCTAssert(_reactSubview == nil, @"Modal view can only have one subview");
   [super insertReactSubview:subview atIndex:atIndex];
   [_touchHandler attachToView:subview];
+#if TARGET_OS_TV
+  if (_onRequestClose) {
```

Comment:
My concern is that if the gesture recognizer is there, but there is no `onRequestClose`, then the application will be stuck with the modal showing.  That said, your statements about needing to override `setOnRequestClose` are correct, and I'll make that change.